src/main/scala/mesosphere/marathon/core/task/update/impl/TaskStatusUpdateProcessorImpl.scala

```diff
@@ -61,15 +62,18 @@ class TaskStatusUpdateProcessorImpl @Inject() (
           mesosStatus = status
         ).map(_ => acknowledge(status))
       case None =>
-        killUnknownTaskTimer {
-          if (status.getState != TaskState.TASK_LOST) {
-            // If we kill a unknown task, we will get another TASK_LOST notification which leads to an endless
-            // stream of kills and TASK_LOST updates.
-            killTask(taskId)
+        if (MesosTaskStatus.Terminal.unapply(status).isDefined) {
```

Comment:
Taste thing: I would either use the extractor instead of calling `unapply` directly:
```scala
status match {
   case MesosTaskStatus.Terminal(_) =>
}
```
or make a function `Terminal.isTerminal(status): Boolean`.